### PR TITLE
Adding xamarin component restore task

### PR DESCRIPTION
**Pull request by chrisrpatterson**

### CODE REVIEW

**Comment by DavidStaheli** [comment]
Looks good to me.

**Comment by madhurig** [comment]
Looks good.

File: Tasks/XamarinComponentRestore/task.loc.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  "description": "ms-resource:loc.description",
+  "helpMarkDown": "ms-resource:loc.helpMarkDown",
+  "category": "Package",
+  "author": "Microsoft Corporation",
+  "version": {
+    "Major": 0,
+    "Minor": 1,
+    "Patch": 0
+  },
+  "minimumAgentVersion": "1.83.0",
```
Comment (madhurig) [inline Tasks/XamarinComponentRestore/task.loc.json]: Version is incorrect, task.loc.json has to be re-generated

File: Tasks/XamarinComponentRestore/icon.svg [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<svg width="203px" height="203px" viewBox="0 0 203 203" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:sketch="http://www.bohemiancoding.com/sketch/ns">
+    <!-- Generator: Sketch 3.2 (9961) - http://www.bohemiancoding.com/sketch -->
```
Comment (madhurig) [inline Tasks/XamarinComponentRestore/icon.svg]: Icon does not show up in the web UI

**Pull request merged**